### PR TITLE
add minimal test for otel agent BYOC Docker build

### PR DESCRIPTION
**Pull request by jackgopack4**
### What does this PR do?
Adds `docker_image_build_otel` integration test to build and run the OTel Agent Bring Your Own Collector (BYOC) Docker image, verifying that a "version" command passed to the otel-agent entrypoint succeeds

### Motivation
Closes [OTEL-1989](https://datadoghq.atlassian.net/browse/OTEL-1989)

### Describe how to test/QA your changes
Run integration_tests stage of gitlab CI, see passing job here: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/662106364

### Possible Drawbacks / Trade-offs
Runs a static command in `otel.yml`, could expand tests and functionality in the future by making the test call an invoke command

### Additional Notes
Also adds updates to JOBOWNERS and CODEOWNERS file to reflect OTEL team ownership of OTEL-related test files.

[OTEL-1989]: https://datadoghq.atlassian.net/browse/OTEL-1989?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by amenasria — APPROVED** [review]
LGTM for Agent DevX Infra owned files !

File: .gitlab/integration_test/otel.yml [hunk 1]
```diff
@@ -21,3 +21,29 @@ integration_tests_otel:
...
+
+docker_image_build_otel:  
+  stage: integration_test
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/docker_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  needs: ["go_deps","integration_tests_otel"]
+  tags: ["runner:docker"]
+  before_script:
+    - !reference [.retrieve_linux_go_deps]
+    - mkdir -p /tmp/otel-ci
+    - cp comp/otelcol/collector-contrib/impl/manifest.yaml /tmp/otel-ci/
```
Comment (dineshg13) [inline .gitlab/integration_test/otel.yml:34]: can u modify manifest.yaml , by adding a component not present in the default build ? either metricstransform processor or something else

**Review by dineshg13 — CHANGES_REQUESTED** [review]

**Review by jennchenn — APPROVED** [review]
approving the change to have otel own the new test being added

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**